### ISSUE TITLE
`fixupInternalLinks` changes bookmarks

### ISSUE THREAD

**Issue by ohjakobsen** [rstudio/shiny-server-js#17]
I have a custom Shiny app that uses bookmarks in anchor elements (`href="#dest"`) to navigate through the app (I'm using a custom function to render a tabbable UI element with Bootstrap 4). On Shiny Server Pro (1.5.10.990), the links are changed by adding the path of the server to the link (`href="http://servername/#dest"`) effectively breaking the functionality. The culprit seems to be the function `fixupInternalLinks()` in `shiny-server-client.min.js`. Is this expected behavior, or should this function leave bookmarks untouched? The same effect can be seen with a bare-bones app using `navbarPage()` with `tabPanel()`, but in this case the functionality does not break although the internal link is still changed.

*Note:* I only get this behavior on Shiny Server Pro when using another location than `/`.

**Comment by jcheng5** [comment]
Thanks for the report. It is expected that anchor links would be transformed, but not expected that this would leave them broken. Would it be possible for you to send a minimal reproducible example, and if not, at least tell us what your tab UI's generated HTML looks like?

**Comment by ohjakobsen** [comment]
It seems like the issue is with using Bootstrap 4. The following code will work when run in RStudio, but breaks on Shiny Server Pro. If I swap out Bootstrap 4.1.3 with the latest release, `util.js` will throw an uncaught DOMException error.

```
library(shiny)

bs4Page <- function(title, ...) {
  
  tagList(
    tags$head(
      tags$link(rel = 'stylesheet', type = 'text/css', href = 'https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/css/bootstrap.min.css'),
      tags$script(src = 'https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js'),
      tags$script(src = 'https://stackpath.bootstrapcdn.com/bootstrap/4.1.3/js/bootstrap.min.js'),
      tags$title(title)
    ),
    list(...)
  )
  
}

ui <- bs4Page(
  
  title = 'Bug test',
  
  list(tags$nav(
    tags$div(
      class = 'nav nav-tabs', id = 'nav-tab', role = 'tablist',
      list(
        tags$a(
          class = 'nav-item nav-link active', id = 'nav-home-tab',
          `data-toggle` = 'tab', href = '#nav-home', role = 'tab',
          `aria-controls` = 'nav-home', `aria-selected` = 'true', 'Home'),
        tags$a(
          class = 'nav-item nav-link', id = 'nav-profile-tab',
          `data-toggle` = 'tab', href = '#nav-profile', role = 'tab',
          `aria-controls` = 'nav-profile', `aria-selected` = 'true', 'Profile'),
        tags$a(
          class = 'nav-item nav-link', id = 'nav-contact-tab',
          `data-toggle` = 'tab', href = '#nav-contact', role = 'tab',
          `aria-controls` = 'nav-contact', `aria-selected` = 'true', 'Contact'))
    )
  ),
  
  tags$div(
    class = 'tab-content', id = 'nav-tabContent',
    list(
      tags$div(
        class = 'tab-pane fade show active', id = 'nav-home', role = 'tabpanel',
        `aria-labelledby` = 'nav-home-tab', 'Lorem ipsum home tab'
      ),
      tags$div(
        class = 'tab-pane fade', id = 'nav-profile', role = 'tabpanel',
        `aria-labelledby` = 'nav-profile-tab', 'Lorem ipsum profile tab'
      ),
      tags$div(
        class = 'tab-pane fade', id = 'nav-contact', role = 'tabpanel',
        `aria-labelledby` = 'nav-contact-tab', 'Lorem ipsum contact tab'
      ))
  ))
  
)

server <- function(input, output, session) {}

shinyApp(ui, server)
```

**Comment by ohjakobsen** [comment]
**Update:** Adding a `data-target` attribute to the anchor elements seems to fix the issue both with Bootstrap 4.1.3 and 4.2.1.
